File: client/src/enum.cpp
#include "enum.h"
#include <string>

std::string getTypeEntityName(ENTITIES ID) {
  switch(ID) {
    case ENTITIES::ENTITY:default:  return "ENTITY";
    case ENTITIES::ZOMBIE:          return "ZOMBIE";
    case ENTITIES::AUTOTURRET:      return "AUTOTURRET";
  }
}

std::string getEntityNameFromId(int id) {
  ENTITIES type = static_cast<ENTITIES>(id);

  return getTypeEntityName(type);
}

File: server/src/conder/enum.ts
export type INPUTS = {
  MOVEMENT: {
    DOWN: boolean;
    UP: boolean;
    LEFT: boolean;
    RIGHT: boolean;
    MOUSEMOVEMENT: boolean;
    MOVING: boolean;
  },
  MOUSE: {
    x: number;
    y: number;
    status: ("defense" | "attack")[]
  };
  KEYBOARD: Map<string, boolean>; // e.g. "w", "a", "s", "d"
}

export enum ENTITIES {
  ENTITY,
  ZOMBIE,

  AUTO_TURRET
}